docs/src/writing_good_rules.md

```diff
@@ -1,5 +1,60 @@
 # On writing good `rrule` / `frule` methods
 
+## Use `Type{T}`, not `typeof(T)`, to define rules for constructors
+
+To define an `frule` or `rrule` for a _function_ `foo` we dispatch on the type of `foo`, which is `typeof(foo)`.
+For example, the `rrule` signature would be like:
+
+```julia
+function rrule(::typeof(foo), args...; kwargs...)
+    ...
+    return y, foo_pullback
+end
+```
+
+But to define an `rrule` for a constructor for a  _type_ `T` we need to be careful to dispatch only on `Type{T}`.
+
+For example, the `rrule` signature for a constructor would be like:
+
+```julia
+function rrule(::Type{T}, args...; kwargs...)
```

Comment:
Might be good to mention when you want `<:T` here, because I've seen this version trip up users before as well.